### PR TITLE
chore(deps): updating packages [Modern JS Guild]

### PR DESCRIPTION
**Pull request by nicholasgriffintn**
## Summary

This PR updates the packages in package.json

**Changes:**

- Added `dist` to gitignore
- Upgraded node to v14
- Changed BBC links to `https`
- Added config for electron-mocha so it sets the right `webPreferences`
- Added `enableRemoteModule: true`, `contextIsolation: false` and `nodeIntegration: true` to `webPreferences`
- Updated cucumber and cucumber/electron to the latest versions, with the new names
- Updated `lib/electron/storage` to match the new functionality in Electron
- Changed the docType test to use `document.doctype` rather than jQuery
- Updated the test command to use the config
- Updated packages as far as possible without breaking existing code too much

Things to note:

- Should we use node 14 or 16? I did have it at 16, but I've thought about it and I think 14 is actually a better minimum.

## Motivation and Context

The majority of the packages are currently out of date, outside of them simply being out of date, this means that we can't use newer features of the browser, alongside this, bbc-a11y currently does not support M1 macbooks, as noted in this issue: https://github.com/bbc/bbc-a11y/issues/345

## How Has This Been Tested?

I have ran all of the tests after upgrading, I had to update one test `test/configSpec.js` as the error now outputs differently

## Types of changes

~~- [ ] Bug fix (non-breaking change which fixes an issue)~~
~~- [ ] New feature (non-breaking change which adds functionality)~~
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I've added tests for my code
~~- [ ] My change requires a change to the documentation.~~
~~- [ ] I have updated the documentation accordingly.~~

### CODE REVIEW

**Comment by nicholasgriffintn** [comment]
Using electron remote resolves the .session is not defined errors, but brings up new ones:

```
Runner
    with a single a page as an argument
WebContents #1 called ipcRenderer.sendSync() with 'REMOTE_BROWSER_GET_BUILTIN' channel without listeners.
      1) checks the URL
    with no arguments
WebContents #1 called ipcRenderer.sendSync() with 'REMOTE_BROWSER_GET_BUILTIN' channel without listeners.
      2) loads the config file


  1) Runner
       with a single a page as an argument
         checks the URL:
     TypeError: Cannot read properties of undefined (reading 'members')
      at setObjectPrototype (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:201:53)
      at metaToValue (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:303:9)
      at Object.getBuiltin (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:360:12)
      at Object.get [as session] (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:398:28)
      at Object.get [as session] (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/index.js:4:80)
      at Object.clear (/Users/nicholasgriffin/Development/bbc-a11y/lib/electron/storage.js:3:48)
      at checkNextPage (/Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:116:20)
      at run (/Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:161:10)
      at /Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:185:14

  2) Runner
       with no arguments
         loads the config file:
     TypeError: Cannot read properties of undefined (reading 'members')
      at setObjectPrototype (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:201:53)
      at metaToValue (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:303:9)
      at Object.getBuiltin (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:360:12)
      at Object.get [as session] (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/remote.js:398:28)
      at Object.get [as session] (/Users/nicholasgriffin/Development/bbc-a11y/node_modules/@electron/remote/dist/src/renderer/index.js:4:80)
      at Object.clear (/Users/nicholasgriffin/Development/bbc-a11y/lib/electron/storage.js:3:48)
      at checkNextPage (/Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:116:20)
      at run (/Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:161:10)
      at /Users/nicholasgriffin/Development/bbc-a11y/lib/runner.js:185:14
```

I'll need to figure out what these errors are, once we do, we should be able to finish this upgrade hopefully.

**Comment by nicholasgriffintn** [comment]
The error that I'm getting, points to this:

https://github.com/electron/electron/issues/31809#issuecomment-1051223721

But I'm pretty sure I've already done that so I'm not sure where this still needs to be done, maybe it's one of the testing libraries?

**Comment by nicholasgriffintn** [comment]
Okay I've got it running now, sadly, I'm still getting errors though, this time with the standards tests themselves, so this may be something to do with another package or browser changes.

**Comment by nicholasgriffintn** [comment]
Okay, everything is passing now aside from this test:

https://github.com/bbc/bbc-a11y/blob/main/features/standards/mag/principles.feature#L154

Which I think is failing because Chrome now handles docType differently? It's possible that pages always have a docType now? Should we just remove this test?

File: features/step_definitions/a11y_steps.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /* eslint no-eval: "off" */
-const { Given, When, Then } = require('cucumber')
+const { Given, When, Then } = require('@cucumber/cucumber')
 const assert = require('assert')
 const Standards = require('../../lib/standards')
 const jquery = require('jquery')
 const webServer = require('../support/web_server')
 
-Given('a website running at http://localhost:{int}', function (port) {
+Given('a website running on locclhost', function (port) {
```
Comment (andymsuk) [inline features/step_definitions/a11y_steps.js:8]: Typo

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:8.16.0-jessie
+FROM node:14.19.3-alpine3.15
```
[the remark this answers is not in the document]
Comment (nicholasgriffintn) [reply docker/Dockerfile:1]: Whoops, forgot that one, just updated it :)

File: features/step_definitions/a11y_steps.js (hunk 1 above)
Comment (nicholasgriffintn) [reply features/step_definitions/a11y_steps.js]: Whoops, my bad, I've sorted this now.

**Review by sbeaumontweb — DISMISSED** [review]

**Comment by sbeaumontweb** [comment]
I think your workaround for the final failure there is probably fine. Let me know if you'd like me to merge this and I'll be happy to if Andy is away. :)

**Comment by nicholasgriffintn** [comment]
Thanks @sbeaumontweb, I'm happy for this to be merged in, @andymsuk is back, I imagine you'll want to manage a release process for this change.

We previously suggested that it might be best to release this as a `canary` version before an official version so that people can check that it is resolving the issue as expected.

**Comment by andymsuk** [comment]
Yep, happy for you to merge and release @sbeaumontweb as a beta/canary release. Thanks!

**Comment by nicholasgriffintn** [comment]
I was wondering the progress on this, it has been a while and bbc-a11y doesn't work at all at the moment really, I have published a version of this here while I wait, as this is becoming more of an issue for us:

https://www.npmjs.com/package/@bbc/bbc-a11y-renovated

As a private repo, this is only available internally as I didn't want to publicise a different version of this, would. be great if this base version could be upgraded though.

**Comment by nicholasgriffintn** [comment]
Seems like the tests no longer work since I last touched this :(

**Pull request closed without merging**